### PR TITLE
Retry support in JAVA Library

### PR DESCRIPTION
**Pull request by NiklasHerrmann21**
- Removed static self-reference
- Implemented thread-safe singleton pattern using double-checked locking
- Updated Javadoc for consistency with default class style
- Added edge case tests for singleton behavior and property injection

### CODE REVIEW

**Review by kuntzed — APPROVED** [review]

File: token-client/src/test/java/com/sap/cloud/security/xsuaa/client/XsuaaOAuth2TokenServiceTest.java [hunk 1]
```diff
@@ -206,9 +206,20 @@ void requestAccessToken_successfulResponse_returnsTokenResponse() throws OAuth2S
...
 
+  @Test
+  public void
+      retrieveTokenKeys_numberFormatExceptionWhileParsingExpiresIn_throwsServiceException() {
+    mockResponse(Map.of(ACCESS_TOKEN, TEST_ACCESS_TOKEN, EXPIRES_IN, "STRING"), 200);
+
+    assertThatThrownBy(() -> cut.requestAccessToken(TOKEN_KEYS_ENDPOINT_URI, headers, PARAMS))
+        .isInstanceOf(OAuth2ServiceException.class)
+        .hasMessageContaining("Invalid expires_in value")
+        .hasMessageContaining(" Response body 'For input string: \"STRING\"'");
```
Comment (kuntzed) [inline token-client/src/test/java/com/sap/cloud/security/xsuaa/client/XsuaaOAuth2TokenServiceTest.java:217]: There is a leading space char.

File: token-client/src/test/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyServiceTest.java [hunk 2]
```diff
@@ -77,24 +82,17 @@ public void retrieveTokenKeys_tokenEndpointUriIsNull_throwsException() {
...
-  @Test
-  public void retrieveTokenKeys_responseNotOk_throwsException() {
-    mockResponse(ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST);
-
-    assertThatThrownBy(() -> cut.retrieveTokenKeys(TOKEN_KEYS_ENDPOINT_URI, PARAMS))
-        .isInstanceOf(OAuth2ServiceException.class)
-        .hasMessageContaining(ERROR_MESSAGE)
-        .hasMessageContaining("Error retrieving token keys");
-  }
-
```
[the remark this answers is not in the document]
Comment (NiklasHerrmann21) [reply token-client/src/test/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyServiceTest.java:89]: Duplicate test as it tests the same as  retrieveTokenKeys_badResponse_throwsException()

**Review by kuntzed — APPROVED** [review]
